x-pack/plugins/security_solution/server/endpoint/routes/policy/handlers.test.ts

```diff
@@ -24,71 +25,207 @@ import { SearchResponse } from 'elasticsearch';
-    const hostPolicyResponseHandler = getHostPolicyResponseHandler({
-      logFactory: loggingSystemMock.create(),
-      service: endpointAppContextService,
-      config: () => Promise.resolve(createMockConfig()),
+  describe('test agent policy summary handler', () => {
+    let mockAgentService: jest.Mocked<AgentService>;
+
+    let agentListResult: {
+      agents: Agent[];
+      total: number;
+      page: number;
+      perPage: number;
+    };
+
+    let emptyAgentListResult: {
+      agents: Agent[];
+      total: number;
+      page: number;
+      perPage: number;
+    };
+
+    beforeEach(() => {
+      mockScopedClient = elasticsearchServiceMock.createLegacyScopedClusterClient();
+      mockSavedObjectClient = savedObjectsClientMock.create();
+      mockResponse = httpServerMock.createResponseFactory();
+      endpointAppContextService = new EndpointAppContextService();
+      mockAgentService = createMockAgentService();
+      emptyAgentListResult = {
+        agents: [],
+        total: 2,
+        page: 1,
+        perPage: 1,
+      };
+
+      agentListResult = {
+        agents: [
+          ({
+            local_metadata: {
+              elastic: {
+                agent: {
+                  version: '8.0.0',
+                },
+              },
+            },
+          } as unknown) as Agent,
+          ({
+            local_metadata: {
+              elastic: {
+                agent: {
+                  version: '8.0.0',
+                },
+              },
+            },
+          } as unknown) as Agent,
+          ({
+            local_metadata: {
+              elastic: {
+                agent: {
+                  version: '8.1.0',
+                },
+              },
+            },
+          } as unknown) as Agent,
+        ],
+        total: 2,
+        page: 1,
+        perPage: 1,
+      };
+      endpointAppContextService.start({
+        ...createMockEndpointAppContextServiceStartContract(),
+        ...{ agentService: mockAgentService },
+      });
     });
 
-    mockScopedClient.callAsCurrentUser.mockImplementationOnce(() =>
-      Promise.resolve(createSearchResponse())
-    );
+    afterEach(() => endpointAppContextService.stop());
+
+    it('should return the summary of all the agent with the given policy name', async () => {
+      mockAgentService.listAgents
+        .mockImplementationOnce(() => Promise.resolve(agentListResult))
+        .mockImplementationOnce(() => Promise.resolve(emptyAgentListResult));
 
-    const mockRequest = httpServerMock.createKibanaRequest({
-      params: { agentId: 'id' },
+      const policySummarysHandler = getAgentPolicySummaryHandler({
+        logFactory: loggingSystemMock.create(),
+        service: endpointAppContextService,
+        config: () => Promise.resolve(createMockConfig()),
+      });
+
+      const mockRequest = httpServerMock.createKibanaRequest({
+        query: { policy_id: 'my-policy', package_name: 'endpoint' },
```

Comment:
nit: if someone were to look at the tests to understand the input/output of the API, and saw policy_id with a string like this, they may think this field is actually for names. Could use a more ID-looking test value here for clarity